devicetypes/keen-home/keen-home-smart-vent.src/keen-home-smart-vent.groovy

```diff
@@ -51,423 +44,113 @@ metadata {
-    ])
-
-    // send a move command to ensure level attribute gets reset for old, buggy firmware
-    // then send a reset to factory defaults
-    // finally re-configure to ensure reports and binding is still properly set after the rtfd
-    [
-        makeLevelCommand(device.currentValue("level")), "delay 500",
-        "st cmd 0x${device.deviceNetworkId} 1 0 0 {}", "delay 5000"
-    ] + configure()
+    zigbee.off()
 }
 
 def setLevel(value, rate = null) {
     log.debug "setting level: ${value}"
-    def linkText = getLinkText(device)
-
-    // only change the level if the vent is not obstructed
-    def currentState = device.currentValue("switch")
-
-    if (currentState == "obstructed") {
-        log.error("cannot set level because ${linkText} is obstructed")
-        return
-    }
-
-    sendEvent(name: "level", value: value)
-    if (value > 0) {
-        sendEvent(name: "switch", value: "on", descriptionText: "${linkText} is on by setting a level")
-    }
-    else {
-        sendEvent(name: "switch", value: "off", descriptionText: "${linkText} is off by setting level to 0")
-    }
-
-    makeLevelCommand(value)
-}
-
-def getOnOff() {
-    log.debug "getOnOff()"
-
-    // disallow on/off updates while vent is obstructed
-    if (device.currentValue("switch") == "obstructed") {
-        log.error("cannot update open/close status because ${getLinkText(device)} is obstructed")
-        return []
-    }
-
-    ["st rattr 0x${device.deviceNetworkId} 1 0x0006 0"]
-}
-
-def getPressure() {
-    log.debug "getPressure()"
-
-    // using a Keen Home specific attribute in the pressure measurement cluster
-    [
-        "zcl mfg-code 0x115B", "delay 200",
-        "zcl global read 0x0403 0x20", "delay 200",
-        "send 0x${device.deviceNetworkId} 1 1", "delay 200"
-    ]
-}
-
-def getLevel() {
-    log.debug "getLevel()"
-
-    // disallow level updates while vent is obstructed
-    if (device.currentValue("switch") == "obstructed") {
-        log.error("cannot update level status because ${getLinkText(device)} is obstructed")
-        return []
-    }
-
-    ["st rattr 0x${device.deviceNetworkId} 1 0x0008 0x0000"]
-}
-
-def getTemperature() {
-    log.debug "getTemperature()"
-
-    ["st rattr 0x${device.deviceNetworkId} 1 0x0402 0"]
-}
-
-def getBattery() {
-    log.debug "getBattery()"
-
-    ["st rattr 0x${device.deviceNetworkId} 1 0x0001 0x0021"]
-}
-
-def setZigBeeIdTile() {
-    log.debug "setZigBeeIdTile() - ${device.zigbeeId}"
-
-    def linkText = getLinkText(device)
-
-    sendEvent([
-        name: "zigbeeId",
-        value: device.zigbeeId,
-        descriptionText: "${linkText} has zigbeeId ${device.zigbeeId}" ])
-    return [
-        name: "zigbeeId",
-        value: device.zigbeeId,
-        descriptionText: "${linkText} has zigbeeId ${device.zigbeeId}" ]
+    def cmds = []
+    cmds << zigbee.setLevel(value)
+    cmds << "delay 1000"
+    cmds << zigbee.levelRefresh()
+    cmds
```

Comment:
Could probably just make this `[zigbee.setLevel(value), "delay 1000", zigbee.levelRefresh()]`